### PR TITLE
Support context propagation for ChannelHandler

### PR DESCRIPTION
**Pull request by violetagg**

### CODE REVIEW

**Comment by violetagg** [comment]
@reactor/netty-team @marcingrzejszczak PTAL

**Review by marcingrzejszczak — APPROVED** [review]

File: reactor-netty-core/src/main/java/reactor/netty/channel/MicrometerChannelMetricsHandler.java [hunk 1]
```diff
@@ -155,7 +151,10 @@ public void connect(ChannelHandlerContext ctx, SocketAddress remoteAddress,
 			       status = future.isSuccess() ? SUCCESS : ERROR;
 
 			       observation.stop();
-			});
+
+			           ReactorNetty.setChannelContext(ctx.channel(), parentContextView);
+			           parentContextView = null;
+			   });
```
Comment (pderop) [inline reactor-netty-core/src/main/java/reactor/netty/channel/MicrometerChannelMetricsHandler.java:157]: nit: tabulation from lines 155,156 are not aligned with the ones above (lines 152,152)

Comment (violetagg) [reply reactor-netty-core/src/main/java/reactor/netty/channel/MicrometerChannelMetricsHandler.java]: fixed, thanks

**Comment by violetagg** [comment]
@pderop @marcingrzejszczak Thanks for the review!

**Pull request merged**